File: app/src/main/java/com/bangnv/cafeorder/model/java/RetrofitClients.java
package com.bangnv.cafeorder.model.java;

import androidx.annotation.NonNull;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClients {
    private static Retrofit retrofit;
//    private static final String URL_REQUEST = "http://103.226.249.210:8239/api/"; // đã sửa theo kiểu fake
    private static final String URL_REQUEST = "http://103.226.249.210:8239/api/";
    @NonNull
    public static Retrofit getInstance() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder().baseUrl(URL_REQUEST).addConverterFactory(GsonConverterFactory.create()).build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/bangnv/cafeorder/model/baseresponse/RetrofitClients.kt
package com.bangnv.cafeorder.model.baseresponse

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClients {

    // Mạng wifi ở nhà: URL_REQUEST
//    private const val URL_REQUEST = "http://192.168.0.103:3006/"
//    private const val URL_REQUEST = "http://192.168.0.102:3006/"

    // Deployed URL_REQUEST
    private const val URL_REQUEST = "https://coffeenbnode.onrender.com/"

    private val retrofit: Retrofit by lazy {
        Retrofit.Builder()
            .baseUrl(URL_REQUEST)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }

    @JvmStatic
    fun getInstance(): Retrofit = retrofit

    @JvmStatic
    fun getMoMoInstance(): Retrofit {
        val moMoUrl = "https://test-payment.momo.vn/"
        return Retrofit.Builder()
            .baseUrl(moMoUrl)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }
}
